src/main/java/org/apache/commons/text/similarity/LongestCommonSubsequence.java

```diff
@@ -0,0 +1,142 @@
+ * of the commonly shared length implies that the strings are completely the same in value and position.
+ * <i>Note.</i>  Generally this algorithm is fairly inefficient, as for length <i>m</i>, <i>n</i> of the input
+ * <code>CharSequence</code>'s <code>left</code> and <code>right</code> respectively, the runtime of the
+ * algoirthm is <i>O(m*n)</i>.
+ * </p>
+ *
+ * <p>
+ * This implementation is based on the Longest Commons Substring algorithm
+ * from <a href="https://en.wikipedia.org/wiki/Longest_common_subsequence_problem">
+ * https://en.wikipedia.org/wiki/Longest_common_subsequence_problem</a>.
+ * </p>
+ *
+ * <p>For further reading see:</p>
+ *
+ * <p>Lothaire, M. <i>Applied combinatorics on words</i>. New York: Cambridge U Press, 2005. <b>12-13</b></p>
+ *
+ * @since 1.0
+ */
+public class LongestCommonSubsequence implements SimilarityScore<Integer> {
+
+    /**
+     * Calculates longestCommonSubsequence similarity scoreof two set character sequence passed as
+     * input.
+     *
+     * @param left first character sequence
+     * @param right second character sequence
+     * @return longestCommonSubsequenceLength
+     * @throws IllegalArgumentException
+     *             if either String input {@code null}
+     */
+    @Override
+    public Integer apply(final CharSequence left, final CharSequence right) {
+        // Quick return for invalid inputs
+        if (left == null || right == null) {
+            throw new IllegalArgumentException("Inputs must not be null");
+        }
+        return logestCommonSubsequence(left, right).length();
+    }
+
+    /**
+     *
+     * Computes the longestCommonSubsequence between the two strings given.
+     *
+     * <p>
+     * Note, a substring and
+     * subsequence are not necessarily the same thing. Indeed, <code>abcxyzqrs</code> and
+     * <code>xyzghfm</code> have both the same common substring and subsequence, namely <code>xyz</code>. However,
+     * <code>axbyczqrs</code> and <code>abcxyzqtv</code> have the longest common subsequence <code>xyzq</code> because a
+     * subsequence need not have adjacent characters.
+     * </p>
+     *
+     * <p>
+     * For reference, we give the definition of a subsequence for the reader: a <i>subsequence</i> is a sequence that can be
+     * derived from another sequence by deleting some elements without changing the order of the remaining elements.
+     * </p>
+     *
+     * @param left
+     * @param right
+     * @return lcsLengthArray
+     * @throws IllegalArgumentException
+     *             if either String input {@code null}
+     */
+    public CharSequence logestCommonSubsequence(final CharSequence left, final CharSequence right) {
+        // Quick return
+        if (left == null || right == null) {
+            throw new IllegalArgumentException("Inputs must not be null");
+        }
+        StringBuffer longestCommonSubstringArray = new StringBuffer(Math.max(left.length(), right.length()));
```

Comment:
Probably makes no difference to have a StringBuilder or StringBuffer here? Or should we use StringBuilder?